versioned/gc/core/src/main/java/org/projectnessie/versioned/gc/CategorizedValue.java

```diff
@@ -0,0 +1,80 @@
+/*
+ * Copyright (C) 2020 Dremio
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.projectnessie.versioned.gc;
+
+import java.io.Serializable;
+
+import com.google.protobuf.ByteString;
+
+/**
+ * Referenced state of a value, its type and its byte[] representation.
+ */
+public final class CategorizedValue implements Serializable {
+
+  private static final long serialVersionUID = -1466847843373432962L;
+
+  private boolean referenced;
+  private byte[] data;
+  private String valueType;
```

Comment:
I'm confused by valueType here. I don't think is a ValueType so what is it? I think maybe you're talking about contents type? What is the thinking on the mapping of this string? Are you proposing canonical strings for contents at the tiered version store level?

Reply:
The thinking behind this was that the identification of AssetKeys from Values is dependent on the value type: eg the iceberg listing is different from Delta. This helps easy identification of which asset key listing algo should be used for this row. The entity type concept being discussed in other PRs would eventually fill this role so in the meantime this uses the simple class name. Happy to remove it until the entity type PRs are closed if you prefer. However I did rename to entity type to make its use more clear.

Reply:
Part of the confusion here was specifically the name ValueType, which has another meaning. In general, I'm not a huge fan of using a string as opposed to an enum.

Reply:
I was going to add an enum but it required knowledge of the `DummyValue` in tests which I didn't like. I see two options: remove entity type from `CategorizedValue` until the entity type changes go into the `VersionStore` or leave it as is and take a second pass after the entity type changes. I am not bothered either way :-)